src/expression.c

```diff
@@ -9571,7 +9573,10 @@ Expression *DeleteExp::semantic(Scope *sc)
                 FuncDeclaration *fd = sd->dtor;
 
                 if (!f)
+                {
+                    semanticTypeInfo(sc, ts);
```

Comment:
I think the compiler needs a larger refactoring: the transformations/lowerings made in the glue layer must happen while semantic analysis is still possible. That would obsolete all this guessing what typeinfo will be needed during code generation, sometimes translated to something else making the typeinfo unnecessary. Without that change templated library implementation of associative and dynamic arrays are unlikely to produce good inlining results.

My current idea regarding such a refactoring would be to do the lowerings in a "semantic4" step, that starts once semantic3 is complete, but can still go back to the earlier stages. That way, user code will produce error messages as now, but low-level library code might show the internal names instead.